### PR TITLE
[MRG] fixed bug in linearized poisson gradient and hessian

### LINKED ISSUE #181: Probable wrong gradient in linearized Poisson case
Hi,
I think the gradient (perhaps Hessian too) of linearized Poisson case is probably wrong in the cheatsheet. Eta in the second sum of gradient wrt beta_0 and beta_j should be exp(eta).

### PR DESCRIPTION
**Pull request by pavanramkumar**
closes #181

### CODE REVIEW

File: pyglmnet/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from .pyglmnet import GLM, set_log_level
 from .utils import softmax, label_binarizer, log_likelihood
 from .datasets import fetch_tikhonov_data
-__version__ = '1.0.0'
+__version__ = '1.0.1'
```
Comment (jasmainak) [inline pyglmnet/__init__.py]: why did you do this?

**Comment by jasmainak** [comment]
@pavanramkumar just a thought. Should we use `autograd` or `theano` in our test suite to avoid such problems in the future? I know that we can also overhaul the entire backend thing, but maybe we go one step at a time?

File: pyglmnet/__init__.py (hunk 1 above)
Comment (pavanramkumar) [inline pyglmnet/__init__.py]: in semver.org the idea is to version as `MAJOR.MINOR.PATCH`. my thought was that until the next PyPI release (which should be called `2.0.0`), we can categorize our PRs into `MINOR` or `PATCH` and accordingly increment the version number. do you have a suggestion?

**Comment by pavanramkumar** [comment]
@jasmainak i love the idea of using theano to do gradient checking but don't have the bandwidth to implement it this year

File: pyglmnet/__init__.py (hunk 1 above)
Comment (pavanramkumar) [inline pyglmnet/__init__.py]: @jasmainak lmk if you think we should use another strategy for versioning. if not, i can merge and update our users!

Comment (jasmainak) [inline pyglmnet/__init__.py]: normally we do `0.1` then `0.2`, upto `0.10` ... you keep going till you feel confident that you can do a `1.0` release. And then for each of these versions, say `0.7`, you can have sometimes a maintenance version like `0.7.1` or `0.7.2` if you screwed up something during your `0.7` release. But normally not more than one or two maintenance versions. And during development, you append `dev0` to the version. So, something like `0.8.dev0` if you're developing for the `0.8` version. See here for the guidelines: https://www.python.org/dev/peps/pep-0440/

**Pull request merged**